src/kpro.erl

```diff
@@ -153,9 +154,15 @@ produce_request(Topic, Partition, KvList,
     #kpro_TopicMessageSet{ topicName             = Topic
                          , partitionMessageSet_L = [PartitionMsgSet]
                          },
+  %% Encode message set here right now.
+  %% Instead of keeping a possibily very large array
+  %% and passing it around processes
+  %% e.g. in brod, the message set can be encoded in producer
+  %% worker before sending it down to socket process
+  MessageSetBin = iolist_to_binary(encode({array, [TopicMessageSet]})),
   #kpro_ProduceRequest{ requiredAcks      = RequiredAcks
                       , timeout           = AckTimeout
-                      , topicMessageSet_L = [TopicMessageSet]
+                      , topicMessageSet_L = {already_encoded, MessageSetBin}
```

Comment:
1. it's more search friendly
2. the encode/1 call is very generic, it doesn't know which specific piece of data is being encoded. 'just binary' would for example clash this clause.
```
encode({string, B}) when is_binary(B) ->
  Length = size(B),
  <<Length:16/?INT, B/binary>>;
```